app/models/topic_converter.rb

```diff
@@ -81,6 +81,7 @@ def add_allowed_users
       user.user_stat.save!
     end
     @topic.topic_allowed_users.build(user_id: @user.id) unless @topic.topic_allowed_users.where(user_id: @user.id).exists?
+    @topic.topic_allowed_users = @topic.topic_allowed_users.uniq(&:user_id)
```

Comment:
does this fix it? Can you add a failing test case here?

Reply:
@SamSaffron fixed - https://github.com/discourse/discourse/pull/8169/commits/5244b3d0e4784606c92e2fed4e1f2f1a273524bf